lib/packages/core/lib/src/utils/media/media_handler.dart

```diff
@@ -0,0 +1,198 @@
+import 'dart:io';
+
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/foundation.dart';
+import 'package:likeminds_chat_fl/likeminds_chat_fl.dart';
+import 'package:likeminds_chat_flutter_core/src/utils/utils.dart';
+import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+
+/// A class to manage all media picking, and accessing
+///
+/// Gives access to functions that handles media picking natively
+class LMChatMediaHandler {
+  /// A function that returns a List of videos from device storage
+  static Future<LMResponse<List<LMChatMediaModel>>> pickVideos(
+      int currentMediaLength) async {
+    try {
+      List<LMChatMediaModel> videoFiles = [];
+      final FilePickerResult? pickedFiles = await FilePicker.platform.pickFiles(
+        allowMultiple: true,
+        type: FileType.video,
+      );
+
+      if (pickedFiles == null || pickedFiles.files.isEmpty) {
+        return LMResponse(success: true);
+      }
+
+      const double sizeLimit = 100;
+
+      for (PlatformFile pFile in pickedFiles.files) {
+        double fileSize = getFileSizeInDouble(pFile.size);
+
+        if (fileSize > sizeLimit) {
+          return LMResponse(
+              success: false,
+              errorMessage:
+                  'Max file size allowed: ${sizeLimit.toStringAsFixed(2)}MB');
+        } else {
+          LMChatMediaModel videoFile;
+          if (kIsWeb) {
+            videoFile = LMChatMediaModel(
+              mediaType: LMChatMediaType.video,
+              mediaFile: File(pFile.path!),
+              size: fileSize.toInt(),
+              duration: 10,
+              meta: {
+                'file_name': pFile.name,
+              },
+            );
+          } else {
+            videoFile = LMChatMediaModel(
+              mediaType: LMChatMediaType.video,
+              mediaFile: File(pFile.path!),
+              size: fileSize.toInt(),
+              duration: 10,
+              meta: {
+                'file_name': pFile.name,
+              },
+            );
+          }
+          videoFiles.add(videoFile);
+        }
+        return LMResponse(success: true, data: videoFiles);
+      }
+      return LMResponse(success: true);
+    } on Exception catch (err) {
+      return LMResponse(
+        success: false,
+        errorMessage: 'An error occurred\n${err.toString()}',
+      );
+    }
+  }
+
+  /// A function that returns an image from device storage
+  static Future<LMResponse<LMChatMediaModel>> pickSingleImage() async {
+    final FilePickerResult? list = await FilePicker.platform.pickFiles(
+      allowMultiple: false,
+      type: FileType.image,
```

Comment:
add `compressionQuality: 0,` here, There is a bug in FilePicker, in which app got crashed on Android:10. This is a workaround for this bug.